File: app/src/main/java/com/example/doan/Network/RoleDeserializer.java
package com.example.doan.Network;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonParseException;
import java.lang.reflect.Type;

public class RoleDeserializer implements JsonDeserializer<String> {
    @Override
    public String deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) 
            throws JsonParseException {
        if (json.isJsonPrimitive()) {
            return json.getAsString();
        } else if (json.isJsonObject()) {
            // Nếu backend trả về object {"name": "MANAGER"}
            return json.getAsJsonObject().get("name").getAsString();
        }
        return null;
    }
}

File: app/src/main/java/com/example/doan/Network/RoleDeserializer.kt
package com.example.doan.Network

import com.google.gson.JsonDeserializationContext
import com.google.gson.JsonDeserializer
import com.google.gson.JsonElement
import com.google.gson.JsonParseException
import java.lang.reflect.Type

class RoleDeserializer : JsonDeserializer<String> {
    
    @Throws(JsonParseException::class)
    override fun deserialize(
        json: JsonElement,
        typeOfT: Type,
        context: JsonDeserializationContext
    ): String? {
        return when {
            json.isJsonPrimitive -> json.asString
            json.isJsonObject -> {
                // Nếu backend trả về object {"name": "MANAGER"}
                json.asJsonObject.get("name").asString
            }
            else -> null
        }
    }
}
